plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryConfig.java

```diff
@@ -88,6 +89,19 @@ public BigQueryConfig setParentProjectId(String parentProjectId)
         return this;
     }
 
+    public Optional<String> getQuotaProjectId()
+    {
+        return quotaProjectId;
+    }
+
+    @Config("bigquery.quota-project-id")
+    @ConfigDescription("The Google Cloud Project ID to attribute quota usage")
```

Comment:
according to the javadocs both billing and quota happens against what is set via `setQuotaProjectId`.

The `setParentProjectId` is not clearly documented.

From all that I read so far it appears that:

- By default the quota project id is taken from the service account credentials which means there's no need to explicitly set it unless user credentials are being used instead of service accounts

So is my understanding correct that this change is related to your other change at https://github.com/trinodb/trino/pull/17717?

This is useful context IMO and maybe it makes more sense to implement both changes in a single PR and explain this background in the commit message. This also needs to be clarified in docs why this config exists otherwise users won't know when or when not to configure this.

Reply:
> So is my understanding correct that this change is related to your other change at https://github.com/trinodb/trino/pull/17717?

Actually, the need for this stems out of the need to correctly attribute quota usage when using the Streaming API for writes. Currently, everything is being attributed to the table underlying project id. 

>By default the quota project id is taken from the service account credentials which means there's no need to explicitly set it unless user credentials are being used instead of service accounts

Couldn't fully confirm this but given the above, it seems that could be implementation specific.
 
That said this is why I feel that this should be a separate PR. Though I feel, given a snd thought that I'm gonna revert my changes and keep quota project id set with the Trino's parent project id.

Reply:
If the goal is to attribute quota and cost having a single config for that will be useful. Is there a case where you'd want different values for parent project id and quota project id?

Reply:
> Is there a case where you'd want different values for parent project id and quota project id?

Yeah that's what I was reflecting about, and I think ideally they should be the same. The name "parent" is just an artifact of the storage api which in essence is the same as the quota project id.

Reply:
>If the goal is to attribute quota and cost having a single config for that will be useful.
+1 so I'm leaning in just keeping the `bigquery.parent-project-id` and probably augment a bit the docs to encompass the quota.